app/services/SubmissionService.scala

```diff
@@ -94,21 +93,30 @@ class SubmissionService @Inject()(
 
   def submit(providerId: String, answers: MovementAnswers)(implicit hc: HeaderCarrier): Future[ConsignmentReferences] = {
     val cache = Cache(providerId, answers)
-
-    val data = movementBuilder.createMovementRequest(providerId, answers)
+    val data = movementBuilder.createMovementExchange(providerId, answers)
     val timer = metrics.startTimer(cache.answers.`type`)
 
-    auditService.auditAllPagesUserInput(answers)
-
-    val movementAuditType =
-      if (cache.answers.`type` == JourneyType.ARRIVE) AuditTypes.AuditArrival else AuditTypes.AuditDeparture
-
-    connector.submit(data).map { _ =>
+    def performSideEffects(result: String)(implicit hc: HeaderCarrier): Unit = {
```

Comment:
performSideEffects??

Reply:
Whatever this is we probably need a more appropriate name & to make it private

Reply:
This is an inner method in `submit`. Don't need to be private. And it is an inner method because it requires 4 parameters which are available in the mentioned method.

Reply:
Good point it's an inner method. In terms of naming maybe something like:
 `onComplete(success)` & `onComplete(failed)`
or
`on(success) & `on(failed)`